### PR TITLE
Fixes bar_with_pessimistic_uncertainty convergence error

### PR DESCRIPTION
**Pull request by badisa**
* A bug introduced in https://github.com/proteneer/timemachine/pull/1324

### CODE REVIEW

File: tests/test_bar.py [hunk 1]
```diff
@@ -180,6 +180,11 @@ def test_df_from_u_kln_does_not_raise_on_incomplete_convergence():
     assert np.isfinite(df)
     assert np.isnan(ddf)  # returns NaN for uncertainty on incomplete convergence
 
+    bootstrap_df, bootstrap_ddf = bar_with_pessimistic_uncertainty(u_kln, maximum_iterations=1)
```
Comment (badisa) [inline tests/test_bar.py:183]: This raises an exception without the fix. Didn't launch a CI job as it is an unambigious failure

File: timemachine/fe/bar.py [hunk 2]
```diff
@@ -179,12 +179,8 @@ def bootstrap_bar(
     * TODO[deboggle] -- upgrade from pymbar3 to pymbar4 and remove this
     * TODO[performance] -- multiprocessing, if needed?
     """
-    u_kn, N_k = ukln_to_ukn(u_kln)
-    mbar = pymbar.MBAR(u_kn, N_k, maximum_iterations=maximum_iterations)
 
-    df, ddf = mbar.getFreeEnergyDifferences()
-    full_bar_result = df[0, 1]
-    full_bar_err = ddf[0, 1]
+    full_bar_result, full_bar_err = df_and_err_from_u_kln(u_kln, maximum_iterations=maximum_iterations)
```
Comment (badisa) [inline timemachine/fe/bar.py:183]: df_and_err_from_u_kln handles the behavior correctly: https://github.com/proteneer/timemachine/blob/master/timemachine/fe/bar.py#L136-L139

**Review by maxentile — APPROVED** [review]

File: timemachine/fe/bar.py [hunk 3]
```diff
@@ -193,6 +189,9 @@ def bootstrap_bar(
     seed = 2022
     rng = np.random.default_rng(seed)
 
+    u_kn, N_k = ukln_to_ukn(u_kln)
+    mbar = pymbar.MBAR(u_kn, N_k, maximum_iterations=maximum_iterations)
```
Comment (mcwitt) [inline timemachine/fe/bar.py:193]: nit: this seems unnecessary vs. setting `initial_f_k=None` below (or maybe better, `initial_f_k=np.array([0.0, df])`)

**Review by mcwitt — APPROVED** [review]
Good catch. Makes sense to reuse `df_and_err_from_u_kln` since the latter has the desired workaround.

File: timemachine/fe/bar.py (hunk 3 above)
Comment (badisa) [reply timemachine/fe/bar.py:193]: Good call, will replace it with `initial_f_k=np.array([0.0, df])` which is the same value without constructing an extra MBAR object

Comment (badisa) [reply timemachine/fe/bar.py]: Set using pre-existing value in https://github.com/proteneer/timemachine/pull/1344/commits/a7de23f8700f0af98ed72ea49ec221fdff85566f

**Pull request merged**